File: dataset/human/cpp/file_168.cpp
#include <iostream>

int main(){
    int R,G,B,N;
    std::cin>>R>>G>>B>>N;
    int ans=0;
    for(int i=0;i<=N;++i){
        for(int j=0;j<=N;++j){
            if( (N-i*R-G*j)>=0 && (N-i*R-G*j)%B==0) ans+=1;
        }

    }
    std::cout<<ans<<std::endl;


}

File: dataset/human/python/file_168.py
r, g, b, n = map(int, input().split())

import math
ans = 0
for i in range(math.ceil(n / r)+1):
    for j in range(math.ceil(n / g)+1):
        if n - r * i - g * j < 0:
            break
        if  (n - r * i - g * j) % b == 0:
            ans += 1
print(ans)
